guide/sections/part1/data-publisher.adoc

```diff
@@ -1,23 +1,204 @@
+
+Finally, you should consider versioning your API to avoid breaking changes when adding new features. A common approach is add a _version number_ prefix into the API path; e.g., ``/v1/service/{rest-of-path}`` or ``/service/v1/{rest-of-path}``.
+
+More guidance on use of interactive APIs in WIS2 is anticipated in future versions of this Guide.
+
+===== Providing data in (near) real-time
+
+WIS2 is designed to support the data sharing needs of all WMO programmes. Among these, the World Weather Watch footnote:[WMO World Weather Watch https://wmo.int/world-weather-watch] drives specific needs for the rapid exchange of data to support weather forecasting.
+
+To enable real-time data sharingfootnote:[In the context of WIS2, real-time implies anything from a few seconds to a few minutes - not the milliseconds required by some applications.], WIS2 uses notification messages to advertise the availability of a new resource - data or discovery metadata - and how to access that resource. Notification messages are published to a queue on a message broker in your WIS2 Nodefootnote:[WIS2 ensures rapid global distribution of notification messages using a network of Global Brokers which subscribe to message brokers of WIS2 Nodes and republish notification messages (see section 2.4.2. Global Broker (TODO CROSSREF)).] using the MQTT protocol and immediately delivered to everyone subscribing to that queue. A queue is associated with a specific _topic_, such as dataset.
+
+For example, when a new temperature profile from a radio sonde deployment is added to a dataset of upper-air data measurements, a notification message would be published that includes the URL used to access the new temperature profile data. Everyone subscribing to notification messages about the upper-air measurement dataset would receive the notification message, identify the URL and download the new temperature profile data.
+
+Optionally, data may be embedded in a notification message using a ``content`` object _in addition_ to publishing via the data server. Embedded data must be encoded as ``UTF-8``, ``Base64``, or ``gzip``, and must not exceed 4096 bytes in length once encoded.
```

Comment:
base64

Embedded -> inline (aligns to WNM language)

Reply:
Fixed.